Add CA certs plug-in guide (#1083)
kyma-project/istio: docs/user/troubleshooting/03-90-istio-cert-unknown.md

# Istio cannot verify HTTPS certificate generated by a trusted signing CA

## Possible symptoms

- Istiod logs include multiple warnings/errors that include `x509: certificate signed by unknown authority`.
- A set up JWT authorization flow does not work, and requests with valid tokens end up with a 401 Status error `Jwt verification fails`.
- Istio sidecars Envoy configuration for JWT is not up-to-date.

> - Checking server certificate chain reveals different Root CAs then for working servers.

Reviewer: typo, should be 'than'

## Cause

By default Istio requires that in case an outbound connection to a HTTPS server is initiated, the server certificate is signed by a trusted Certificate Authority, which means that at least the Root CA certificate presented by the server must be recognized by the client.
If, for some reason the server uses an urecognized Root CA to generate it's domain certificate (e.g. the server uses internal solution specific Root CA or it migrated to a fairly new, not widely recognized Root CA), the HTTPS handshake will fail, and the connection will not be established.

> This might, for example, cause the JWKS fetch from the JWKS URI to fail.

Reviewer 2: ```suggestion
As a result, the JWKS fetch from the JWKS URI may fail.
```

## Remedy

> To ensure that the certificate is trusted by Istio, make sure that you are using the most up-to-date version of the Istio module. If you are on BTP Kyma Runtime this is guaranteed by the solution.

Reviewer 2: ```suggestion
To ensure that the certificate is trusted by Istio, verify that you are using the most up-to-date version of the Istio module.  If you are using BTP Kyma Runtime, the solution guarantees that you have the most up-to-date version. 
```

> # Istio cannot verify HTTPS certificate generated by a trusted signing CA

Reviewer 2: ```suggestion
# Istio Cannot Verify an HTTPS certificate Generated by a Trusted Signing CA
```

## Symptom

See the possible symptoms:
Istiod logs include multiple warnings or errors containing the message `x509: certificate signed by unknown authority`.
- The JWT authorization flow does not function correctly, and requests with valid tokens result in a `401` status error, `Jwt verification fails.`
- Istio sidecars Envoy configuration for JWT is not up-to-date.
- Checking the server certificate chain reveals different root Certificate Authorities (CAs) than working servers.

## Cause

By default, Istio requires that if an outbound connection to an HTTPS server is initiated, the server certificate must be signed by a trusted CA. This means that the client must recognize at least the root CA presented by the server.
If the server uses an unrecognized root CA to generate its domain certificate, the HTTPS handshake fails, and the connection is not established. This issue might happen, for example, when the server uses an internal solution-specific root CA or it has been migrated to a new root CA that is not widely recognized.
As a result, the JWKS fetch from the JWKS URI may fail.

## Remedy

> To ensure that the certificate is trusted by Istio, verify that you are using the most up-to-date version of the Istio module.  If you are using BTP Kyma Runtime, the solution guarantees that you have the most up-to-date version. 

Reviewer 2: ```suggestion
To ensure that the certificate is trusted by Istio, verify that you are using the most up-to-date version of the Istio module.  If you are using SAP BTP, Kyma runtime, the solution guarantees that you have the most up-to-date version. 
```
